include/picongpu/fields/incidentField/profiles/DispersivePulse.hpp

```diff
@@ -215,8 +215,11 @@ namespace picongpu
                             return mag;
                         }
 
+                        template<uint32_t T_dim>
                         HDINLINE float_X
-                        phi(floatD_X const& totalCellIdx, float_X const Omega, float_X const phaseShift) const
+                        phi(math::Vector<float_X, T_dim> const& totalCellIdx,
```

Comment:
You can stick with the `floatD_X` type, since totalCellIdx is only used to obtain the `pos` which takes `float2_X` and `float3_X`.

(by the way, you also did not change the signature of the `amp()` in line 152.)

Reply:
We call `phi` from B field calculation always with float3_X now. Therefore `floatD_X` is wrong but you are right we need to change `amp()` too.

Reply:
fixed